plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse_test.go

```diff
@@ -0,0 +1,288 @@
+				Log:          testutil.Logger{},
+			}
+			token, err := s.tokenManager.requestAuthToken()
+			require.NoError(t, err)
+			if token.AccessToken != test.token.AccessToken {
+				t.Errorf("test %v: expect access_toke %v, but get %v\n", test.res, test.token.AccessToken, token.AccessToken)
+			}
+			if token.TokenType != test.token.TokenType {
+				t.Errorf("test %v: expect token_type %v, but get %v\n", test.res, test.token.TokenType, token.TokenType)
+			}
+		})
+	}
+}
+
+const path = "/automation/api/v2/events"
+
+var multiEntries = false
+var mux sync.Mutex
+
+func setMultiEntries(m bool) {
+	mux.Lock()
+	defer mux.Unlock()
+	multiEntries = m
+}
+
+func getMultiEntries() bool {
+	mux.Lock()
+	defer mux.Unlock()
+	return multiEntries
+}
+
+func TestCtrlXCreateSubscriptionBasic(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusCreated)
+		_, err := w.Write([]byte("201 created"))
+		require.NoError(t, err)
+	}))
+	defer server.Close()
+
+	subs := make([]Subscription, 0)
+	subs = append(subs, Subscription{
+		Nodes: []string{"plc/app/Application/sym/PLC_PRG/counter",
+			"plc/app/Application/sym/PLC_PRG/counterReverse"},
+	},
+	)
+	s := &CtrlXDataLayerSSE{
+		connection:   &http.Client{},
+		url:          server.URL,
+		Username:     "user",
+		Password:     "password",
+		tokenManager: tokenManager{url: server.URL, username: "user", password: "password", connection: &http.Client{}, log: testutil.Logger{}},
+		Subscription: subs,
+		Log:          testutil.Logger{},
+	}
+
+	subID, err := s.createSubscription(&subs[0])
+
+	require.NotEmpty(t, subID)
+	require.Equal(t, nil, err)
+}
+
+func TestCtrlXCreateSubscriptionDriven(t *testing.T) {
+	var tests = []struct {
+		res    string
+		status int
+		subID  string
+	}{
+		{res: "{\"status\":200}", status: 200, subID: "subID"},
+		{res: "{\"status\":401}", status: 401, subID: ""},
+	}
+
+	for _, test := range tests {
+		t.Run(test.res, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(test.status)
+				_, err := w.Write([]byte(test.res))
+				require.NoError(t, err)
+			}))
+			defer server.Close()
+			subs := make([]Subscription, 0)
+			subs = append(subs, Subscription{
+				Nodes: []string{"plc/app/Application/sym/PLC_PRG/counter",
+					"plc/app/Application/sym/PLC_PRG/counterReverse"},
+			},
+			)
+			s := &CtrlXDataLayerSSE{
+				connection:   &http.Client{},
+				url:          server.URL,
+				Username:     "user",
+				Password:     "password",
+				Subscription: subs,
+				tokenManager: tokenManager{url: server.URL, username: "user", password: "password", connection: &http.Client{}, log: testutil.Logger{}},
+				Log:          testutil.Logger{},
+			}
+			subID, err := s.createSubscription(&subs[0])
+
+			if test.subID != "" && err == nil {
+				require.NotEmpty(t, subID)
+			} else {
+				require.Empty(t, subID)
+			}
```

Comment:
Is this really the intended behavior? Or is it
```suggestion

			if test.subID != "" {
				require.NoError(t, err)
				require.NotEmpty(t, subID)
			} else {
				require.Error(t, err)
				require.Empty(t, subID)
			}
```

Reply:
Done